src/node_url.cc

```diff
@@ -829,6 +829,16 @@ static url_host_type ParseOpaqueHost(url_host* host,
   return type;
 }
 
+static inline bool IsAllASCII(std::string* input) {
```

Comment:
@mscdex  : folding both checking and lowercasing will result
 either 
1. Code duplication as ContainsNonAscii is used at one more place.  or 
2. Or have a function pointer as argument, its a problem with separation of concern. IIRC since function pointers are of statically know function it will get inlined.

Whats your opinion on this?.

I made changes to re-use ContainsNonAscii. Have a look at that when you get time.